addons/core/translations/resources/en-us.yaml

```diff
@@ -88,6 +89,15 @@ host-catalog:
     unknown: Unknown
     aws: AWS
     azure: Azure
+    dynamic: Dynamic
+  form:
+    provider-details: 
+      label: Provider Details
+    types:
```

Comment:
The pattern for resource-specific types makes put it immediately under the resource.  It's a sibling to `form`, rather than nested under it.